monitoring/zabbix_host.py

```
@@ -356,6 +362,28 @@ def link_or_clear_template(self, host_id, template_id_list):
         except Exception, e:
             self._module.fail_json(msg="Failed to link template to host: %s" % e)
 
+    # Update the host inventory_mode
+    def update_inventory_mode(self, host_id, inventory_mode):
+
+	if not inventory_mode: 
+            return
+
+        if inventory_mode == "automatic":
+            inventory_mode = int(1)
+        elif inventory_mode == "manual":
+            inventory_mode = int(0)
+        else: # inventory_mode == "disabled":
+            inventory_mode = int(-1)
+
+
+        # watch for - https://support.zabbix.com/browse/ZBX-6033
+        request_str = {'hostid': host_id, 'inventory_mode': inventory_mode}
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            self._zapi.host.update(request_str)
+        except Exception, e:
```

catch all exception should be avoided.